microsoft/electionguard-python
# 🗳️ #586 CLI Import Ballots

### Issue

Fixes #586
Fixed #355 

### Description
Adds a new command to the CLI for importing and tallying ballots and decrypting spoiled ballots, and decrypting ballots.  This does not yet include election record, that functionality is in ticket https://github.com/microsoft/electionguard-python/issues/605.

### Testing

1. Run an end-to-end election like:

`poetry run eg e2e --guardian-count=2 --quorum=2 --manifest=data/election_manifest_simple.json --ballots=data/plaintext_ballots_simple.json --spoil-id=25a7111b-4334-425a-87c1-f7a49f42b3a2 --output-record="./election_record.zip" --output-keys="./guardian_private_data.json"`

2. Pull out the ballots, context, and manifest and plunk them in `\data\simple`
3. Run the import command like: `poetry run eg import-ballots --manifest=data/simple/manifest.json --context=data/simple/context.json --ballots-dir=data/simple/ballots --guardian-keys=./guardian_private_data.json` (or `make eg-import-simple-election`)

Expected results:

```
----------------------------------------
Retrieving Inputs
----------------------------------------

----------------------------------------
Retrieving manifest
----------------------------------------
Name: Jefferson County Spring Primary
Scope: jefferson-county-primary
Geopolitical Units: 4
Parties: 3
Candidates: 6
Contests: 2
Ballot Styles: 4
importing submitted_ballot_03a29d15-667c-4ac8-afd7-549f19b8e4eb.json
importing submitted_ballot_1048ce32-f1b1-4b05-b7fb-8c615ac842ee.json
importing submitted_ballot_25a7111b-4334-425a-87c1-f7a49f42b3a2.json
importing submitted_ballot_5a150c74-a2cb-47f6-b575-165ba8a4ce53.json
importing submitted_ballot_69aeacb4-64c6-4205-9bb2-5fb6b3b3ea58.json
importing submitted_ballot_9fee0e77-cfd2-401a-a210-93bbc4dd30ef.json
Ballots Dir: [...]/data/simple/ballots

----------------------------------------
Building election
----------------------------------------

----------------------------------------
Creating Tally
----------------------------------------
Ballots in tally: 6

----------------------------------------
Decrypting tally
----------------------------------------
Cast ballots: 5
Spoiled ballots: 1
Total ballots: 6
Guardian Present: 1
Guardian Present: 2
Lagrange coefficients retrieved: 2

----------------------------------------
Decrypted tally
----------------------------------------
Contest: justice-supreme-court
  john-adams-selection: 5 votes
  benjamin-franklin-selection: 0 votes
  john-hancock-selection: 2 votes
  write-in-selection: 3 votes
Contest: referendum-pineapple
  referendum-pineapple-affirmative-selection: 1 votes
  referendum-pineapple-negative-selection: 1 votes

----------------------------------------
Spoiled ballot '25a7111b-4334-425a-87c1-f7a49f42b3a2'
----------------------------------------
Contest: justice-supreme-court
  john-adams-selection: 1 votes
  benjamin-franklin-selection: 1 votes
  john-hancock-selection: 0 votes
  write-in-selection: 0 votes
  justice-supreme-court-4-placeholder: 0 votes
  justice-supreme-court-5-placeholder: 0 votes
```
